steering/src/main.cpp

```diff
@@ -1,16 +1,75 @@
-#include "Arduino.h"
+#include <Arduino.h>
+#include <SPI.h>
+
 #include "steering_io.h"
-#include "cal.h"
 
-int ledState = 0;
+#include "cal.hpp"
+#include "can.hpp"
+
+#include <EVE.h>
+#include "tft_data.h"
+#include "tft.h"
+
+#include "screens.h"
+
+// Only do print statements if we are in debug mode
+#define STEERING_DEBUG 1
 
 void setup() {
-  Serial2.begin(115200);
-  pinMode(STEERING_STATUS_LED, OUTPUT);           // status LED output
+    Serial2.begin(115200);
+    pinMode(EVE_CS, OUTPUT);
+    digitalWrite(EVE_CS, HIGH);
+    pinMode(EVE_PDN, OUTPUT);
+    digitalWrite(EVE_PDN, LOW);
+
+    pinMode(STEERING_CAN_OE, OUTPUT);
+    digitalWrite(STEERING_CAN_OE, LOW);
+
+    bool ret = CANInit(CAN_1000KBPS, 0, 2);
+    if (!ret) while (true);
+
+    SPI.begin(); // Set up the SPI to run in Mode 0 and 8 MHz
+    SPI.beginTransaction(SPISettings(1000000, MSBFIRST, SPI_MODE0));
+
+    TFT_init();
+
+    // Display splash logo
+    TFT_splash();
+    delay(3000);
 }
 
 void loop() {
-  ledState = !ledState;
-  digitalWrite(STEERING_STATUS_LED, ledState);
-  delay(200);
+    static uint32_t previous_millis = 0;
+    static uint8_t display_delay = 0;
+    uint32_t current_millis;
+    current_millis = millis();
+
+    if ((current_millis - previous_millis) > 3) {
+        // Execute touch display code every ~5 ms
+        previous_millis = current_millis;
+
+        TFT_touch();
+
+        display_delay++;
+        if (display_delay > 3) { // Execute display refresh every ~20ms
+            TFT_display((tft_callback)display_code);
```

Comment:
TODO: Try and get this on a timer interrupt. Can be a separate PR.

Reply:
Let's make an issue for this and resolve for now?